cluster/gce/gci/configure-helper.sh

```diff
@@ -1660,7 +1683,13 @@ function prepare-konnectivity-server-manifest {
   params+=("--uds-name=/etc/srv/kubernetes/konnectivity-server/konnectivity-server.socket")
   params+=("--cluster-cert=/etc/srv/kubernetes/pki/apiserver.crt")
   params+=("--cluster-key=/etc/srv/kubernetes/pki/apiserver.key")
-  params+=("--mode=grpc")
+  if [[ "${KONNECTIVITY_SERVICE_PROXY_PROTOCOL_MODE:-grpc}" == 'http-connect' ]]; then
+    params+=("--mode=http-connect")
+  else
```

Comment:
Please reverse the order (grpc first), add an else if for the http-connect and have a final else which gives an error and exits.
Eg. https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh#L830